src/squeeze_xen.ml

```diff
@@ -370,9 +375,10 @@ module Domain = struct
 end
 
 (** Mark the domain as squeezed declared stuck *)
-let declare_domain_stuck ~xc domid =
+let declare_domain_stuck ~xc domid value =
 	let cnx = xc in
-	Domain.set_domain_stuck_noexn cnx domid "true"
+	let can_balloon = Domain.get_feature_balloon cnx domid in
+	if can_balloon then Domain.set_domain_stuck_noexn cnx domid value
```

Comment:
So the `declare_domain_stuck` function now does not always mark a domain as "stuck". Would it make more sense to have to conditional at the point where `declare_domain_stuck` is called? Or rename the function.